extensions/positron-r/src/conda-activation.ts

```diff
@@ -0,0 +1,155 @@
+	} catch {
+		// Mamba not available, try conda
+		try {
+			await execPromise('conda --version');
+			LOGGER.info('Found conda for environment activation');
+			return CondaCommand.CONDA;
+		} catch {
+			LOGGER.warn('Neither conda nor mamba found in PATH');
+			return undefined;
+		}
+	}
+}
+
+/**
+ * Get environment variables from activating a conda environment
+ *
+ * This function activates the conda environment and captures the resulting
+ * environment variables, which can then be passed to the R kernel process.
+ *
+ * @param condaEnvPath The path to the conda environment to activate
+ * @returns A record of environment variables, or undefined if activation fails
+ */
+export async function getCondaActivationEnvironment(
+	condaEnvPath: string
+): Promise<Record<string, string> | undefined> {
+	const condaCommand = await findCondaCommand();
+	if (!condaCommand) {
+		LOGGER.error('Cannot activate conda environment: conda/mamba not found in PATH');
+		return undefined;
+	}
+
+	try {
+		LOGGER.info(`Activating conda environment at: ${condaEnvPath}`);
+
+		let command: string;
+		if (process.platform === 'win32') {
+			// On Windows, use cmd.exe to activate and print environment
+			// We use && to chain commands so the second only runs if first succeeds
+			command = `cmd /c "${condaCommand} activate ${condaEnvPath} && set"`;
+		} else {
+			// On Unix-like systems, we need to source the conda setup and activate
+			// The key is to source conda.sh (or mamba.sh) first, then activate, then print env
+			const shell = process.env.SHELL || '/bin/bash';
+			const shellName = path.basename(shell);
+
+			// Get conda/mamba base path
+			const { stdout: condaInfo } = await execPromise(`${condaCommand} info --json`);
+			const info = JSON.parse(condaInfo);
+			const condaBasePath = info.root_prefix || info.conda_prefix;
+
+			if (!condaBasePath) {
+				LOGGER.error('Could not determine conda base path');
+				return undefined;
+			}
+
+			// Construct path to activation script
+			let activationScript: string;
+			if (shellName.includes('fish')) {
+				activationScript = path.join(condaBasePath, 'etc', 'fish', 'conf.d', `${condaCommand}.fish`);
+				command = `fish -c "source ${activationScript}; conda activate ${condaEnvPath}; env"`;
+			} else if (shellName.includes('zsh')) {
+				activationScript = path.join(condaBasePath, 'etc', 'profile.d', `${condaCommand}.sh`);
+				command = `zsh -c '. ${activationScript}; conda activate ${condaEnvPath}; env'`;
+			} else {
+				// Default to bash
+				activationScript = path.join(condaBasePath, 'etc', 'profile.d', `${condaCommand}.sh`);
+				command = `bash -c '. ${activationScript}; conda activate ${condaEnvPath}; env'`;
+			}
+
+			LOGGER.debug(`Using activation command: ${command}`);
+		}
+
+		// Execute the command to get the environment
+		const { stdout } = await execPromise(command, {
+			maxBuffer: 10 * 1024 * 1024, // 10MB buffer for large environments
+			timeout: 30000 // 30 second timeout
```

Comment:
If there's any chance that this is going to take > 2 seconds then we should show some kind of progress toast while we're doing it.